### PR TITLE
Fix iDeal support

### PR DESCRIPTION
**Pull request by oursodon**
This feature fix the support for iDeal.

# Changes
+ Added MoneyInIDealFactory
+ Added moneyInIDealInit and MoneyInIDealConfirm to Wallet
+ Fixed formatting and validation of payload

### CODE REVIEW

File: test/wallet/money-in-sdd.js [hunk 1]
```diff
@@ -51,6 +51,9 @@ describe('money in sdd', function () {
       });
     }).then(function (transaction) {
       return done();
-    }).catch(done);
+    }).catch((err) => {
+      console.log(err); // Url returned by lemonway on the dev environment fail to validate SSD Mandate
```
Comment (patateviolente) [inline test/wallet/money-in-sdd.js]: This does not help, no?
I can't open the link, but let's keep away from this PR on iDEAL anyway

File: lib/client.js [hunk 2]
```diff
@@ -440,15 +440,21 @@ Client.prototype.createGiftCodeAmazon = function (ip, _opts) {
...
+  var opts = _.assign({}, _opts, {
+    version: '1.0',
+    amountTot: utils.numberToFixed(_opts.amount),
+    amountCom: utils.numberToFixed(_opts.commission),
+    autoCommission: utils.boolToString(_opts.autoCommission),
+    returnUrl: _opts.returnUrl
+  });
 
-  return this._request('MonetInIDealInit', ip, opts).get('IDEALINIT');
+  return this._request('MoneyInIDealInit', ip, opts).get('IDEALINIT');
```
Comment (jguepin) [inline lib/client.js]: image

File: lib/factories/money-in-iDeal-factory.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+var _ = require('lodash');
+var querystring = require('querystring');
```
Comment (jguepin) [inline lib/factories/money-in-iDeal-factory.js]: We don't use that.

File: lib/factories/wallet-factory.js [hunk 4]
```diff
@@ -148,6 +148,27 @@ function WalletFactory (lemonway) {
...
   };
 
+  this.moneyInIDealInit = function (ip, wallet, _opts) {
+    var opts = _.assign({}, _opts, {
+      wallet: wallet.id ? wallet.id : wallet
+    });
+
+    return lemonway._client.moneyInIDealInit(ip, opts)
+      .bind(this)
+      .then(data => new lemonway.MoneyInIDeal.MoneyInIDeal(data));
```
Comment (jguepin) [inline lib/factories/wallet-factory.js]: We might want to avoid arrow functions here, this would be a breaking change.

**Pull request merged**